mozetl/taar/taar_similarity.py

```diff
@@ -65,20 +82,33 @@ def get_addons_per_client(users_df, addon_whitelist, minimum_addons_count):
 
     def is_valid_addon(guid, addon):
         return not (
-            addon.is_system or
-            addon.app_disabled or
-            addon.type != "extension" or
-            addon.user_disabled or
-            addon.foreign_install or
-            guid not in addon_whitelist
+            addon.is_system
+            or addon.app_disabled
+            or addon.type != "extension"
+            or addon.user_disabled
+            or addon.foreign_install
+            or guid not in addon_whitelist
         )
 
     # Create an add-ons dataset un-nesting the add-on map from each
     # user to a list of add-on GUIDs. Also filter undesired add-ons.
+
+    # Note that this list comprehension was restructured
+    # from the original longitudinal query.  In particular, note that
+    # each client's 'active_addons' entry is a list containing the
+    # a dictionary of {addon_guid: {addon_metadata_dict}}
+
+    def flatten_valid_guid_generator(p):
+        for addon_data in p["active_addons"]:
+            for guid, data in addon_data.items():
+                if not is_valid_addon(guid, data):
+                    continue
+                yield guid
+
     return (
-        users_df.rdd
```

Comment:
cc/ @mlopatka  

Based on the test changes in https://github.com/mozilla/python_mozetl/pull/265/files#r227487130 , I reworked the way that valid addon GUIDs are flattened into a simple list.

Reply:
For reference - the longitudinal table had an array of struct's in the `active_addons` field - probably to denote the set of addons installed over time.